contracts/RecoveryQuorum.sol

```diff
@@ -3,111 +3,138 @@ import "./RecoverableController.sol";
-      controller.changeUserKeyFromRecovery(newUserKey);
-      for(uint i = 0 ; i < delegateAddresses.length ; i++){
-        //remove any pending delegates after a recovery
-        if(delegates[delegateAddresses[i]].pendingUntil > now){ 
-            delegates[delegateAddresses[i]].deletedAfter = now;
+
+    function collectedSignatures(address _proposedUserKey) constant returns (uint signatures) {
+        for(uint i = 0 ; i < delegateAddresses.length ; i++) {
+            if (delegateHasValidSignature(delegates[delegateAddresses[i]])
+                    && delegates[delegateAddresses[i]].proposedUserKey == _proposedUserKey) {
+                signatures++;
+            }
         }
-        delete delegates[delegateAddresses[i]].proposedUserKey;
-      }
     }
-  }
 
-  function replaceDelegates(address[] delegatesToRemove, address[] delegatesToAdd) onlyUserKey{
-    for(uint i = 0 ; i < delegatesToRemove.length ; i++){
-      removeDelegate(delegatesToRemove[i]);
+    function getAddresses() constant returns (address[]) {
+        return delegateAddresses;
     }
-    garbageCollect();
-    for(uint j = 0 ; j < delegatesToAdd.length ; j++){
-      addDelegate(delegatesToAdd[j]);
+
+    function neededSignatures() constant returns (uint) {
+        uint currentDelegateCount; //always 0 at this point
+        for(uint i = 0 ; i < delegateAddresses.length ; i++) {
+            if(delegateIsCurrent(delegates[delegateAddresses[i]])) {
+                currentDelegateCount++;
+            }
+        }
+        return currentDelegateCount/2 + 1;
+    }
+
+    function addDelegate(address delegate) private {
+        if(!delegateRecordExists(delegates[delegate])
+                && delegateAddresses.length < MAX_DELEGATES) {
+            delegates[delegate] = Delegate({
+                proposedUserKey: 0x0,
+                pendingUntil: now + controller.longTimeLock(),
+                deletedAfter: NEVER
+            });
+            delegateAddresses.push(delegate);
+        }
     }
-    RecoveryEvent("replaceDelegates", msg.sender);
-  }
-  function collectedSignatures(address _proposedUserKey) returns (uint signatures){
-    for(uint i = 0 ; i < delegateAddresses.length ; i++){
-      if (delegateHasValidSignature(delegates[delegateAddresses[i]]) && delegates[delegateAddresses[i]].proposedUserKey == _proposedUserKey){
-        signatures++;
-      }
+
+    function removeDelegate(address delegate) private {
+        if(delegates[delegate].deletedAfter > controller.longTimeLock() + now) {
+            //remove right away if they are still pending
+            if(delegates[delegate].pendingUntil > now) {
+                delegates[delegate].deletedAfter = now;
+            } else {
+                delegates[delegate].deletedAfter = controller.longTimeLock() + now;
+            }
+        }
     }
-  }
 
-  function getAddresses() constant returns (address[]){ return delegateAddresses; }
+    function garbageCollect() private {
```

Comment:
Yeah, I agree. And I think you are right, not worth worrying about as we will probably move to SSR.